core/src/services/compfs/backend.rs

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use super::core::CompioThread;
+use crate::raw::*;
+use crate::*;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use std::path::PathBuf;
+
+/// [`compio`]-based file system support.
+#[derive(Debug, Clone, Default)]
+pub struct CompFSBuilder {
+    root: Option<PathBuf>,
+}
+
+impl CompFSBuilder {
+    /// Set root for CompFS
+    pub fn root(&mut self, root: &str) -> &mut Self {
+        self.root = if root.is_empty() {
+            None
+        } else {
+            Some(PathBuf::from(root))
+        };
+
+        self
+    }
+}
+
+impl Builder for CompFSBuilder {
+    const SCHEME: Scheme = Scheme::CompFS;
```

Comment:
opendal has strict naming rules for service name.

Please use consist naming every where:

- `compfs`, `Compfs`

Or

- `comp_fs`, `comp-fs`, `CompFs`

or

- `comp_f_s`, `comp-f-s`, `CompFS`